### PR TITLE
switch to micros

### PR DESCRIPTION
**Pull request by floriantschopp**

### CODE REVIEW

**Comment by floriantschopp** [comment]
Still some serious problems with the sync, other functions might also use nh.now() (e.g. data send sync)!

**Comment by floriantschopp** [comment]
Working now :-)
Related PR to compile messages [siemens_tools/#43](https://github.com/ethz-asl/siemens_tools/pull/43)

figure_1
figure_2
figure_3

File: rosserial_client/src/ros_lib/ros/node_handle.h [hunk 1]
```diff
@@ -373,37 +374,38 @@ class NodeHandle_ : public NodeHandleBase_
...
 
   Time now()
   {
-    uint32_t ms = hardware_.time();
+    uint64_t mus = hardware_.time_micros();
     Time current_time;
-    current_time.sec = ms / 1000 + sec_offset;
-    current_time.nsec = (ms % 1000) * 1000000UL + nsec_offset;
+    current_time.sec = mus / 1000000 + sec_offset;
+    current_time.nsec = (mus % 1000000) * 1000000UL + nsec_offset;
```
Comment (andrepfr) [inline rosserial_client/src/ros_lib/ros/node_handle.h]: maybe my brain is already in friday evening mode... but shouldn't this be 
```
 current_time.nsec = (mus % 1000000) * 1000UL + nsec_offset;
```

File: rosserial_client/src/ros_lib/ros/node_handle.h [hunk 2]
```diff
@@ -373,37 +374,38 @@ class NodeHandle_ : public NodeHandleBase_
...
   }
 
   void setNow(Time & new_now)
   {
-    uint32_t ms = hardware_.time();
-    sec_offset = new_now.sec - ms / 1000 - 1;
-    nsec_offset = new_now.nsec - (ms % 1000) * 1000000UL + 1000000000UL;
+    uint32_t mus = hardware_.time_micros();
+    sec_offset = new_now.sec - mus / 1000000 - 1;
+    nsec_offset = new_now.nsec - (mus % 1000000) * 1000000UL + 1000000000UL;
```
Comment (andrepfr) [inline rosserial_client/src/ros_lib/ros/node_handle.h]: same here as above

File: rosserial_client/src/ros_lib/ros/node_handle.h (hunk 1 above)
Comment (floriantschopp) [reply rosserial_client/src/ros_lib/ros/node_handle.h]: You might be right! Thanks for this catch. Unfortunately, I can not test it right away as I don't have my cameras atm...

**Pull request merged**